### PR TITLE
add google analytics script, fix #423

### LINKED ISSUE #423: Poser un google analytics sur -www.cafebabel.com-
ETQ Matt je veux consulter le traffic pour vérifier l'audience et éclairer les décisions de CFB

- [x] doit permettre de moucharder l'ensemble des pages non privatives
- [x] doit permettre de moucharder l'ensemble des pages privatives

note : valider le site par google (fichier à poser à la racine) permettrait d'utiliser : https://search.google.com

### PR DESCRIPTION
**Pull request by johangiraud**

### CODE REVIEW

File: cafebabel/templates/base.html [hunk 1]
```diff
@@ -135,4 +135,13 @@ <h4>Want our bi-monthly newsletter in your inbox? Then type in your email addres
...
 {% endassets %}
 {% block extra_js %}{% endblock %}
+<!-- Global site tag (gtag.js) - Google Analytics -->
+<script async src="https://www.googletagmanager.com/gtag/js?id=UA-126606-5"></script>
+<script>
+  window.dataLayer = window.dataLayer || [];
+  function gtag(){dataLayer.push(arguments);}
+  gtag('js', new Date());
+
+  gtag('config', 'UA-126606-5');
```
Comment (davidbgk) [inline cafebabel/templates/base.html]: That UA string should come from the config

File: cafebabel/templates/base.html [hunk 2]
```diff
@@ -135,4 +135,13 @@ <h4>Want our bi-monthly newsletter in your inbox? Then type in your email addres
     <script src={{ ASSET_URL }}></script>
 {% endassets %}
 {% block extra_js %}{% endblock %}
+<!-- Global site tag (gtag.js) - Google Analytics -->
+<script async src="https://www.googletagmanager.com/gtag/js?id=UA-126606-5"></script>
```
Comment (davidbgk) [inline cafebabel/templates/base.html]: Same here.

File: cafebabel/templates/base.html [hunk 3]
```diff
@@ -135,4 +135,13 @@ <h4>Want our bi-monthly newsletter in your inbox? Then type in your email addres
...
+<script>
+  window.dataLayer = window.dataLayer || [];
```
Comment (davidbgk) [inline cafebabel/templates/base.html]: You can remove semi-colons everywhere for consistency 😉

**Review by davidbgk — CHANGES_REQUESTED** [review]

**Comment by vinyll** [comment]
@davidbgk the code is now in config. Can you review again pls?

File: cafebabel/__init__.py [hunk 4]
```diff
@@ -190,5 +190,6 @@ def add_template_helpers():
             lang_url_for=helpers.lang_url_for,
             absolute=helpers.absolute,
             static_pages_for=helpers.static_pages_for,
-            social_network_url_for=helpers.social_network_url_for
+            social_network_url_for=helpers.social_network_url_for,
+            google_analytics_id=app.config.get('GOOGLE_ANALYTICS_ID'),
```
Comment (davidbgk) [inline cafebabel/__init__.py]: You don't need that, `config` is available globally so `config.GOOGLE_ANALYTICS_ID` will work in templates.

File: cafebabel/tests/test_base.py [hunk 5]
```diff
@@ -98,3 +98,8 @@ def error():
     response = client.get('/foobar/')
     assert response.status_code == HTTPStatus.INTERNAL_SERVER_ERROR
     assert '500 error' in response
+
+
+def test_analytics_in_layout(client):
+    response = client.get('/', follow_redirects=True)
+    assert '/gtag/js?id=testing-analytics">' in response
```
Comment (davidbgk) [inline cafebabel/tests/test_base.py:105]: 👍

**Review by davidbgk — APPROVED** [review]
See comments

**Pull request merged**